src/bot/index.ts

```diff
@@ -41,71 +34,50 @@ export async function startBot(): Promise<void> {
     return;
   }
 
-  const twitchIrcClient = await Promise.all(
-    channels.map(async (ch) => {
-      const user = await prisma.webuiUser.findFirst({
-        where: { channelId: ch.id },
-      });
-
-      if (!user) {
-        return;
-      }
-
-      return new Promise<TwitchIrc>((res) => {
-        const twitchClientForChannel = new TwitchIrc(
-          ircClient,
-          `#${ch.name}`,
-          user.id,
-          async ({ channel }) => {
-            if (apis[channel]) {
-              return;
-            }
-
-            logger.info(`Creting Twitch API connector for channel: ${channel}`);
-            const api = await chatters(channel, mainOAuthToken);
-
-            if (api) {
-              apis[channel] = api;
-              res(twitchClientForChannel);
-            }
-          },
-        );
-      });
-    }),
-  );
-
-  for (const client of twitchIrcClient.filter(Boolean)) {
-    await setDefaultCommandsForChannel(client.channel);
-
-    client.onMessage(async ({ channel, tags, message, self }, it) => {
-      if (self) {
-        return;
-      }
-
-      const handler = await getChatHandler({
-        channel,
-        tags,
-        message,
-        settings: it.settings,
-        api: apis[channel],
-      });
-
-      for (const h of handler) {
-        await h.useHandler({
-          channel,
-          tags,
-          message,
-          client,
-          settings: it.settings,
-        });
-      }
+  async function updateConnection(ch: Channel) {
+    const user = await prisma.webuiUser.findFirst({ where: { channelId: ch.id } });
+    if (!user || !!connections[ch.name] || !ircClient || !mainOAuthToken) {
+      return;
+    }
+
+    connections[ch.name] = new ChannelConnection({
+      ircClient,
+      channelName: `#${ch.name}`,
+      ownerId: user.id,
+      authToken: mainOAuthToken,
     });
+
+    await connections[ch.name].setup();
+  }
+
+  logger.info(`Setting up initial connections: ${channels.length}`);
+  for (const ch of channels) {
+    await updateConnection(ch);
   }
 
+  logger.info("Setting interval to refresh channels list");
+  setInterval(async () => {
+    const res = await prisma.channel.findMany({ where: { enabled: true } });
+
+    for (const ch of res) {
+      await updateConnection(ch);
+    }
+
+    // TODO: Disable connections, not only add new
+    // for (const con of Object.keys(connections)) {
+    //   if (mappedIds.includes(con)) {
+    //     continue;
+    //   }
+    //
+    //   logger.info(`Disabling connection for channel: ${con}`);
+    //   delete connections[con];
+    // }
+  }, 30000);
+
   console.timeEnd("bootstrap");
 }
 
 startBot();
 
 // NOTE: Keep Alive xD
-setInterval(() => {}, 100000);
+setInterval(() => {}, 100_000);
```

Comment:
find better solution

Reply:
it's working with bun --watch / bun dev but once I use bun start, it's just exists after running `startBot();`